### ISSUE TITLE
ShippingRate response model does not match Recharge API

### ISSUE THREAD

**Issue by jonmckinley** [mattgenious/RechargeSharp#15]
The [developer API for retrieving shipping rates](https://developer.rechargepayments.com/v1#retrieve-shipping-rates-for-a-checkout) for a checkout shows `phone_required` as a boolean value, however my testing reveals that this property is possibly null.

This causes a Newtonsoft JSON deserialization exception:
```
Newtonsoft.Json.JsonSerializationException: Error converting value {null} to type 'System.Boolean'. Path 'shipping_rates[0].phone_required', line 1, position 229.   ---> System.InvalidCastException: Null object cannot be converted to a value type.     at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)     --- End of inner exception stack trace ---     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateList(IList list, JsonReader reader, JsonArrayContract contract, JsonProperty containerProperty, String id)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)     at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)     at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)     at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)     at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)     at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)     at RechargeSharp.Services.Checkouts.CheckoutService.RetrieveShippingRatesAsync(String token, OverrideShippingLinesRequest overrideShippingLinesRequest)     at Amora.Promotions.Web.Infrastructure.Services.ApiRequestServiceExtensions.LogAsync[TResponse](IApiRequestService service, ApiRequestTypeEnum requestType, HttpMethod method, String url, Func`1 requestFunc, Object request, Order order) in C:\Users\Jon\source\repos\***\Services\ApiRequestServiceExtensions.cs:line 48
```

As a resolution to this issue, I propose changing `ShippingRate.PhoneRequired` to a `bool?` type. I intend to submit a PR to resolve this issue shortly.

Here is an example Postman request showing the property as `null`:

![image](https://user-images.githubusercontent.com/87319620/125304031-3f346880-e2fb-11eb-8c56-7ae17ead4205.png)

**Comment by mattgenious** [comment]
Thank you for your contribution, I will merge and release tonight. As this is breaking change I will bump the major version, even if it feels a little silly for such a small change.
I also found a bunch of these inconsistencies in their documentation but I think it stems from their backend being python, which means their code is dynamically typed 😊

**Comment by jonmckinley** [comment]
Awesome! Thank you, Matt!

**Comment by mattgenious** [comment]
Closing as you solved it with #16  :-)
